### PR TITLE
Update check benefit financial support condition logic

### PR DESCRIPTION
**Pull request by edwardkerry**
This PR updates the check-benefits-support data file, reordering results to the desired order, removing and adding some additional results, and updating the logical conditions in line with latest guidance.

### CODE REVIEW

File: test/unit/calculators/checks_benefits_support_calculator_test.rb [hunk 1]
```diff
@@ -325,57 +325,32 @@ class CheckBenefitsSupportCalculatorTest < ActiveSupport::TestCase
         end
       end
 
-      context "#eligible_for_30hrs_free_childcare_3_4yrs_scotland?" do
-        should "return true if eligible for 30 Hours Free Childcare for 3 and 4 Year Olds Scotland" do
+      context "# eligible_for_funded_early_learning_and_childcare?" do
```
Comment (Rosa-Fox) [inline test/unit/calculators/checks_benefits_support_calculator_test.rb:552]: nitpick... there is a space between the # and e

File: lib/smart_answer/calculators/check_benefits_support_calculator.rb [hunk 2]
```diff
@@ -83,7 +83,8 @@ def eligible_for_15hrs_free_childcare_3_4yr_olds?
     end
 
     def eligible_for_30hrs_free_childcare_3_4yrs?
-      @are_you_working == "yes_over_16_hours_per_week" &&
+      %w[england northern-ireland].include?(@where_do_you_live) &&
+        @are_you_working != "no" &&
         @children_living_with_you == "yes" &&
         @age_of_children.split(",").any?("3_to_4")
```
[the remark this answers is not in the document]
Comment (edwardkerry) [reply lib/smart_answer/calculators/check_benefits_support_calculator.rb:129]: :)

File: config/smart_answers/check_benefits_support_data.yml [hunk 3]
```diff
@@ -14,6 +14,13 @@
       url: /jobseekers-allowance
       url_text: Anchor link text
       group: Supporting your income
+    - name: jobseekers_allowance_northern_ireland
+      title: Jobseeker's Allowance (JSA)
+      description:
+      condition: eligible_for_jobseekers_allowance_northern_ireland?
+      url: https://docs.google.com/spreadsheets/d/1sXjNndB1QSHmVMZVAyIvrnEsjLQJwAnfp7oZpXEfBpc/edit#gid=0
```
Comment (Rosa-Fox) [inline config/smart_answers/check_benefits_support_data.yml:28]: Wrong url!

File: lib/smart_answer/calculators/check_benefits_support_calculator.rb [hunk 4]
```diff
@@ -26,7 +26,15 @@ def number_of_benefits
     end
 
     def eligible_for_employment_and_support_allowance?
-      @over_state_pension_age == "no" &&
+      @where_do_you_live != "northern-ireland" &&
+        @over_state_pension_age == "no" &&
+        @disability_or_health_condition == "yes" &&
+        disability_affecting_work != "no"
```
Comment (Rosa-Fox) [inline lib/smart_answer/calculators/check_benefits_support_calculator.rb:32]: should be @disability_affecting_work instance var

File: lib/smart_answer/calculators/check_benefits_support_calculator.rb [hunk 5]
```diff
@@ -118,35 +153,70 @@ def eligible_for_child_disability_payment_scotland?
...
+      @children_living_with_you == "yes" &&
+        @age_of_children.split(",").any? { |age| eligible_child_ages.include?(age) }
+    end
+
+    def eligible_for_personal_independence_payment_northern_ireland?
+      return unless @where_do_you_live == "northern-ireland" && @over_state_pension_age == "no"
+
+      return true if @disability_or_health_condition == "yes"
+
+      eligible_child_ages = %w[16_to_17 18_to_19]
```
Comment (Rosa-Fox) [inline lib/smart_answer/calculators/check_benefits_support_calculator.rb:180]: We don't need this for NI as it applies to any age.

File: lib/smart_answer/calculators/check_benefits_support_calculator.rb [hunk 6]
```diff
@@ -118,35 +153,70 @@ def eligible_for_child_disability_payment_scotland?
...
+      @children_living_with_you == "yes" &&
+        @age_of_children.split(",").any? { |age| eligible_child_ages.include?(age) }
     end
 
     def eligible_for_attendance_allowance?
-      @over_state_pension_age == "no" && disability_or_health_condition == "yes"
+      @over_state_pension_age == "yes" && disability_or_health_condition == "yes"
```
Comment (Rosa-Fox) [inline lib/smart_answer/calculators/check_benefits_support_calculator.rb:186]: disability_or_health_condition -> @disability_or_health_condition

**Review by Rosa-Fox — APPROVED** [review]
👍 nice one!

**Pull request merged**